### PR TITLE
feat: adding nkey private seed property and auth handler

### PR DESCRIPTION
**Pull request by agaita**
- Added NKey private seed property.
- Using static credentials auth handler when NKey is present.
- Updating dependencies versions (jnats, spring boot, spring cloud stream, etc.)

### CODE REVIEW

File: nats-spring-boot-starter/pom.xml [hunk 1]
```diff
@@ -25,12 +25,12 @@
 		<dependency>
 			<groupId>io.nats</groupId>
 			<artifactId>jnats</artifactId>
-			<version>2.5.2</version>
+			<version>2.11.5</version>
```
Comment (scottf) [inline nats-spring-boot-starter/pom.xml:28]: 2.11.6 would be better

**Review by scottf — APPROVED** [review]
I think it's fine to use 0.4.1 instead of 0.4.1-SNAPSHOT
And then the one note about 2.11.6 instead of 2.11.5 
Other than the version stuff this LGTM

File: nats-spring-boot-starter/pom.xml (hunk 1 above)
Comment (agaita) [reply nats-spring-boot-starter/pom.xml]: Done!

**Comment by agaita** [comment]
@scottf done, you can re-review the PR again

**Pull request merged**